indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -170,10 +174,11 @@ public int compare(Pair<Task, TaskLock> left, Pair<Task, TaskLock> right)
   public TaskLock lock(final Task task, final Interval interval) throws InterruptedException
   {
     giant.lock();
-
     try {
+      if(!activeTasks.contains(task.getId())){
```

Comment:
there's no need to do this here, tryLock will take care of it. ```lock``` is supposed to be an extremely small wrapper.